sr_utilities_common/scripts/sr_utilities_common/conditional_delayed_rostool.py

```diff
@@ -25,30 +25,39 @@ class RosElementsHandler(object):
     def __init__(self, element_type, required_elements_list):
         self._element_type = element_type
         self._required_elements_list = required_elements_list
+        self._found_elements = []
         self.missing_elements = []
 
-    def execute(self):
-        all_elements_available = False
-        roscore_published_elements = self._retrieve_available_elements(self._element_type)
-        self._check_if_element_is_available(roscore_published_elements, self._required_elements_list)
-        if 0 == len(self._required_elements_list):
-            all_elements_available = True
-        self.missing_elements = self._required_elements_list
-        return all_elements_available
-
-    def _retrieve_available_elements(self, object_type):
-        if object_type == "topic":
+    def check_if_required_element_is_available(self):
+        roscore_published_elements = self._retrieve_available_elements()
+        if not isinstance(self._required_elements_list, list):
+            raise ValueError("{}: Required elements must be a list not a string, \
+                             check the required element list declaration".format(rospy.get_name()))
+        if self._required_elements_list:
+            for element in self._required_elements_list:
+                if element and type(element) == str:
+                    if any(element in sublist for sublist in roscore_published_elements):
+                        rospy.loginfo("Found %s", element)
+                        self._found_elements.append(element)
+                else:
+                    raise ValueError("{}: Required element is not a string".format(rospy.get_name()))
+            if len(self._found_elements) == len(self._required_elements_list):
+                return True
+            self.missing_elements = list(set(self._required_elements_list) - set(self._found_elements))
+        else:
+            return True
+        return False
+
+    def _retrieve_available_elements(self):
+        if self._element_type == "topic":
```

Comment:
please change those to format
"topic" == self._element_type to avoid accidental assignment